src/libtls/tls_server.c

```diff
@@ -190,7 +190,9 @@ static array_t *create_common_key_types(chunk_t hashsig,
 			   reader->read_uint16(reader, &sig_scheme))
 		{
 			lookup = tls_signature_scheme_to_key_type(sig_scheme);
-			if (v == lookup)
+			if (v == lookup ||
+				(v == KEY_ECDSA && lookup == KEY_ED25519) ||
+				(v == KEY_ECDSA && lookup == KEY_ED448))
```

Comment:
If EdDSA is supported locally, shouldn't one or both of these key types be enumerated in the outer enumerator too? This might even cause false positives if ECDSA is supported but EdDSA is not (in particular Ed448 because some plugins only support Ed25519).